src/main/java/de/tum/in/www1/artemis/service/compass/CompassCalculationEngine.java

```diff
@@ -40,31 +42,39 @@
 
     private LocalDateTime lastUsed;
 
-    CompassCalculationEngine(Set<ModelingSubmission> manuallyAssessedSubmissions) {
+    CompassCalculationEngine(Set<ModelingSubmission> modelingSubmissions) {
         lastUsed = LocalDateTime.now();
         modelIndex = new ModelIndex();
         assessmentIndex = new AssessmentIndex();
-
         automaticAssessmentController = new AutomaticAssessmentController();
-        modelSelector = new ModelSelector(); // TODO MJ fix Bug where on load of exercise no
-        // modelsWaitingForAssessment are added ? No differentiation between
-        // submitted and saved assessments!
+        modelSelector = new ModelSelector();
 
-        for (Submission manuallyAssessedSubmission : manuallyAssessedSubmissions) {
+        for (Submission submission : modelingSubmissions) {
             // We have to unproxy here as sometimes the Submission is a Hibernate proxy resulting in a cast exception
             // when iterating over the ModelingSubmissions directly (i.e. for (ModelingSubmission submission : submissions)).
-            ModelingSubmission manuallyAssessedModelingSubmission = (ModelingSubmission) Hibernate.unproxy(manuallyAssessedSubmission);
-            String model = manuallyAssessedModelingSubmission.getModel();
+            ModelingSubmission modelingSubmission = (ModelingSubmission) Hibernate.unproxy(submission);
+
+            String model = modelingSubmission.getModel();
             if (model != null) {
-                buildModel(manuallyAssessedModelingSubmission);
-                buildAssessment(manuallyAssessedModelingSubmission);
-                modelSelector.addAlreadyAssessedModel(manuallyAssessedModelingSubmission.getId());
+                // build the model and add it to Compass
+                buildModel(modelingSubmission);
```

Comment:
What does buildModel mean here?

Reply:
I removed the line comment above the method call here and added proper JavaDoc to the method definition itself.